src/stores/auth.js

```diff
@@ -0,0 +1,73 @@
+import { defineStore } from "pinia";
+import * as authApi from "@/api/services/Authentication.js";
+import { removeAuthHeader, setAuthHeader } from "@/api/axiosConfig.js";
+
+const TOKEN_STORAGE_KEY = "auth-token";
+const USER_STORAGE_KEY = "user";
+
+export const useAuthStore = defineStore({
+  id: "auth",
+  state: () => ({
+    userData: null,
+    authToken: null,
+  }),
+  getters: {
+    isLoggedIn: (state) => !!state.authToken,
+    isUserLoggedIn: (state) => {
+      return state.authToken ? state.userData.role === "user" : "";
+    },
+    isEmployeeLoggedIn: (state) => {
+      return state.authToken ? state.userData.role === "employee" : "";
+    },
+  },
+  actions: {
+    setUserData({ authToken, user }) {
+      this.authToken = authToken;
+      this.userData = user;
+      localStorage.setItem(TOKEN_STORAGE_KEY, authToken);
+      localStorage.setItem(USER_STORAGE_KEY, JSON.stringify(user));
+    },
+    resetUserData() {
+      this.authToken = null;
+      this.userData = null;
+      localStorage.removeItem(TOKEN_STORAGE_KEY);
+      localStorage.removeItem(USER_STORAGE_KEY);
+    },
+    restoreUserData() {
+      try {
+        const authToken = localStorage.getItem(TOKEN_STORAGE_KEY);
+        const user = JSON.parse(localStorage.getItem(USER_STORAGE_KEY) || null);
+        if (authToken && user) {
+          setAuthHeader(`Bearer ${authToken}`);
+          this.setUserData({ authToken, user });
+        }
+      } catch (error) {
+        console.error(error);
+        this.resetUserData();
+      }
+    },
+    async login(credentials) {
+      if (this.isLoggedIn) await this.logout();
+      const response = await authApi.login(credentials);
+      const authHeader = response.headers.authorization;
+      const authToken = authHeader.slice("Bearer ".length);
+      setAuthHeader(authHeader);
+      this.setUserData({ authToken, user: response.data });
+    },
+    async register(credentials) {
+      if (this.isLoggedIn) await this.logout();
```

Comment:
This solution was recommended to us by Artur: https://github.com/rossinek/vue3-auth-playground/blob/main/src/store/auth.js
If you recommend some other way please provide some example.

Reply:
I don't have the full context, maybe there is something that I m not aware of. Is there a possibility to log in if you are already logged in? In most case scenarios it wouldn't be possible that's why I m asking.  In a typical case scenario, if you are logged in, all content related (login button, login form, etc) is hidden away from the user. I'm just trying to understand, what purpose it serves. Can you point to a moment when this condition will be met?

Reply:
Pages: homepage, movies, movie, screenings, contact, logi and register are public, only the header changes depending on the fact if the user is logged in. If the user is logged in  header does not have button login and register button, but my account. If the user enter the route "/login" by typing in the browser he will enter the page and will be able to login in, so that there is a function to logout the previous user.

Reply:
Shouldn't the logged-in user be redirected (via navigation guards) in this case? It seems more natural, but since there are no guards this code is valuable. Without your explanation, I would not know :)